go/tools/bazel/runfiles/resolver.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runfiles
+
+import (
+	"errors"
+	"os"
+)
+
+const (
+	RUNFILES_MANIFEST_FILE = "RUNFILES_MANIFEST_FILE"
+	RUNFILES_DIR           = "RUNFILES_DIR"
+)
+
+// These are the possible Runfiles-related errors.
+var (
+	ErrNoRunfilesEnv   = errors.New("runfiles environment missing")
+	ErrManifestInvalid = errors.New("runfiles manifest syntax error")
+)
+
+// Resolver is an interface for a resolver that can take a runfiles path and
+// resolve it to a path on disk.
+type Resolver interface {
+	Resolve(string) (string, error)
```

Comment:
I think it's better to return `(string, bool)` instead of `(string, error)`. In the case where a file is not found in a manifest, it's not really an error, it's just not found (and the caller will try a different variant of the path).

Reply:
SGTM, Done. For context, originally I had the manifest impl checking actual file existence and the directory impl resolving symlinks. I rebuked on this but never changed the interface back to booleans.